feature/src/vif.c

```diff
@@ -394,3 +394,159 @@ int vif(int (*read_frame)(float *ref_data, float *main_data, float *temp_data, i
+    {
+        goto fail_or_end;
+    }
+
+    data_sz = (size_t)stride * h;
+
+    if (!(ref_buf = aligned_malloc(data_sz, MAX_ALIGN)))
+    {
+        printf("error: aligned_malloc failed for ref_buf.\n");
+        fflush(stdout);
+        goto fail_or_end;
+    }
+    if (!(ref_diff_buf = aligned_malloc(data_sz, MAX_ALIGN)))
+    {
+        printf("error: aligned_malloc failed for ref_diff_buf.\n");
+        fflush(stdout);
+        goto fail_or_end;
+    }
+    if (!(prev_ref_buf = aligned_malloc(data_sz, MAX_ALIGN)))
+    {
+        printf("error: aligned_malloc failed for prev_ref_buf.\n");
+        fflush(stdout);
+        goto fail_or_end;
+    }
+    if (!(dis_buf = aligned_malloc(data_sz, MAX_ALIGN)))
+    {
+        printf("error: aligned_malloc failed for dis_buf.\n");
+        fflush(stdout);
+        goto fail_or_end;
+    }
+    if (!(dis_diff_buf = aligned_malloc(data_sz, MAX_ALIGN)))
+    {
+        printf("error: aligned_malloc failed for dis_diff_buf.\n");
+        fflush(stdout);
+        goto fail_or_end;
+    }
+    if (!(prev_dis_buf = aligned_malloc(data_sz, MAX_ALIGN)))
+    {
+        printf("error: aligned_malloc failed for prev_dis_buf.\n");
+        fflush(stdout);
+        goto fail_or_end;
+    }
+    if (!(temp_buf = aligned_malloc(data_sz * 2, MAX_ALIGN)))
+    {
+        printf("error: aligned_malloc failed for temp_buf.\n");
+        fflush(stdout);
+        goto fail_or_end;
+    }
+
+    int frm_idx = 0;
+    while (1)
+    {
+        ret = read_frame(ref_buf, dis_buf, temp_buf, stride, user_data);
+
+        if(ret == 1){
+            goto fail_or_end;
+        }
+        if (ret == 2)
+        {
+            break;
+        }
+
+        // ===============================================================
+        // offset pixel by OPT_RANGE_PIXEL_OFFSET
+        // ===============================================================
+        offset_image(ref_buf, OPT_RANGE_PIXEL_OFFSET, w, h, stride);
+        offset_image(dis_buf, OPT_RANGE_PIXEL_OFFSET, w, h, stride);
+
+        if (frm_idx > 0)
+        {
+            apply_frame_differencing(ref_buf, prev_ref_buf, ref_diff_buf, w, h, stride / sizeof(float));
+            apply_frame_differencing(dis_buf, prev_dis_buf, dis_diff_buf, w, h, stride / sizeof(float));
+		}
+
+        memcpy(prev_ref_buf, ref_buf, data_sz);
+        memcpy(prev_dis_buf, dis_buf, data_sz);
+
+        if (frm_idx == 0)
+        // pay attention for first frame, frame difference for first frame is not well-defined when subtracting the current frame from the previous one
+        // why not define the frame-difference in the opposite direction? Since quality is mostly affected by recency, it is better to get an
+        // unreliable T-VIF SCORE for earlier frames. Therefore we set the scores to some default value (e.g. 0).
+		{
+		    score = 0.0;
+		    score_num = 0.0;
+		    score_den = 0.0 + 1e-5;
+		    for(int scale=0; scale<4; scale++){
+		    	scores[2*scale] = 0.0;
```

Comment:
Always add space around *, =, etc.

Reply:
indeed I try to follow this; however, some pieces of the existing code did not follow it